Source/WTF/wtf/FileSystem.cpp

```diff
@@ -833,6 +833,33 @@ String pathByAppendingComponents(StringView path, const Vector<StringView>& comp
 
 #endif
 
+#if !OS(WINDOWS) && !PLATFORM(COCOA)
+
+String createTemporaryDirectory()
+{
+    std::error_code ec;
+    std::string tempDir = std::filesystem::temp_directory_path(ec);
+    if (ec)
+        return String();
+
+    std::string newTempDirTemplate = tempDir;
+    newTempDirTemplate += "XXXXXXXX";
+
+    char* newTempDir = new char[newTempDirTemplate.size()];
```

Comment:
Maybe we could use a Vector<char> to avoid explicit memory management?

Reply:
Thank you for the feedback. I made the changes to the pull request.